cmd/kube-proxy/app/server.go

```diff
@@ -359,17 +370,22 @@ func (s *ProxyServer) Run() error {
 
 	s.Broadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: s.EventClient.Events("")})
 
-	// Start up a webserver if requested
-	if s.Config.HealthzPort > 0 {
+	// Start up a healthz server if requested
+	if s.healthzServer != nil {
+		s.healthzServer.Run()
+	}
+
+	// Start up a metrics server if requested
+	if s.Config.MetricsPort > 0 {
 		http.HandleFunc("/proxyMode", func(w http.ResponseWriter, r *http.Request) {
 			fmt.Fprintf(w, "%s", s.ProxyMode)
 		})
 		http.Handle("/metrics", prometheus.Handler())
 		configz.InstallHandler(http.DefaultServeMux)
 		go wait.Until(func() {
-			err := http.ListenAndServe(s.Config.HealthzBindAddress+":"+strconv.Itoa(int(s.Config.HealthzPort)), nil)
+			err := http.ListenAndServe(s.Config.MetricsBindAddress+":"+strconv.Itoa(int(s.Config.MetricsPort)), nil)
```

Comment:
Probably have the BindAddress be a string that includes "addr:port". All of the Go libraries consume this and the having it as a string gives people the option to bind to a Unix domain socket, etc